ncbi_genome_download/core.py

```diff
@@ -227,28 +227,33 @@ def download(**kwargs):
             metadata_table.write(get_table_header())
 
     # Actual logic
-    try:
-        download_jobs = []
-        for group in groups:
-            download_jobs.extend(
-                _download(section, group, uri, output, file_format, assembly_level, genus,
-                          species_taxid, taxid, human_readable, refseq_category, table))
 
-        pool = Pool(processes=parallel)
-        jobs = pool.map_async(worker, download_jobs)
-        try:
-            # 0xFFFF is just "a really long time"
-            jobs.get(0xFFFF)
-        except KeyboardInterrupt:  # pragma: no cover
-            # TODO: Actually test this once I figure out how to do this in py.test
-            logging.error("Interrupted by user")
-            return 1
+    def download_jobs_groups():
+        for group in groups:
+            for job in _download(section, group, uri, output, file_format, assembly_level, genus,
+                                 species_taxid, taxid, human_readable, refseq_category, table):
+                yield job
 
+    try:
+        if parallel == 1:
+            for job in download_jobs_groups():
+                worker(job)
+        else:
+            pool = Pool(processes=parallel)
+            jobs = pool.map_async(worker, download_jobs)
```

Comment:
I don't think this will work anymore. Have you tried parallel downloads with your changes? This is always a bit tricky to see looking at diffs, but I'm missing where `download_jobs` is defined now.

Reply:
Yeah I pulled this out of a fork with more changes and it might have gotten lost.